src/auth/src/credentials/user_credential.rs

```diff
@@ -78,11 +82,46 @@ mod test {
         let mut mock = MockTokenProvider::new();
         mock.expect_get_token()
             .times(1)
-            .return_once(|| Err(CredentialError::new(false, BoxError::from("fail"))));
+            .return_once(|| Err(CredentialError::new(false, Box::from("fail"))));
 
         let mut uc = UserCredential {
             token_provider: mock,
         };
         assert!(uc.get_token().await.is_err());
     }
+
+    #[tokio::test]
+    async fn get_headers_success() {
+        let token = Token {
+            token: "test-token".to_string(),
+            token_type: "Bearer".to_string(),
+            expires_at: None,
+            metadata: None,
+        };
+
+        let mut mock = MockTokenProvider::new();
+        mock.expect_get_token().times(1).return_once(|| Ok(token));
+
+        let mut uc = UserCredential {
+            token_provider: mock,
+        };
+        let actual = uc.get_headers().await.unwrap();
+        assert_eq!(actual.len(), 1);
+        assert_eq!(actual[0].0, AUTHORIZATION);
+        assert_eq!(actual[0].1, "Bearer test-token");
+        assert!(actual[0].1.is_sensitive());
```

Comment:
Consider something like:

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=af731026c583586602b056c07e2b44b4

You may need to transform the `HeaderMap` into something with strings...

Reply:
> You may need to transform the `HeaderMap` into something with strings...

Done.

The conversion into the struct is a bit gross (aesthetically), but I find the code easy to read, so I am happy with it.